apps/heft/src/plugins/TypescriptPlugin/privateTypings/ExtendedTypescript.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
```

Comment:
API Extractor calls this file [TypeScriptInternals.ts](https://github.com/microsoft/rushstack/blob/a30cdf5b3caed87c63ee55f2c91dc1409fa7d29b/apps/api-extractor/src/analyzer/TypeScriptInternals.ts#L16).  

I think that's a more intuitive name than ExtendedTypeScript.ts, if this file is really just a bunch of compiler internals.

(BTW I don't see much value in sharing compiler internals declarations between API Extractor and Heft)